CraftingRevisions/ModUserBlueprintData.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Il2Cpp;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace CraftingRevisions
+{
+	internal sealed class ModUserBlueprintData
+	{
+		/// <summary>
+		/// optional name, used for debugging
+		/// </summary>
+		public string? Name { get; set; } = null;
+		public List<ModRequiredGearItem> RequiredGear { get; set; } = new();
+		public string? RequiredTool { get; set; } = null;
+		public List<string>? OptionalTools { get; set; } = new();
+		public string? CraftedResult { get; set; } = null;
+		public int CraftedResultCount { get; set; } = 0;
+		public int DurationMinutes { get; set; } = 0;
+		public string? CraftingAudio { get; set; } = null;
+		public float KeroseneLitersRequired { get; set; } = 0;
+		public float GunpowderKGRequired { get; set; } = 0;
+		public bool RequiresLight { get; set; } = false;
+		public bool Locked { get; set; } = false;
+		public bool IgnoreLockInSurvival { get; set; } = true;
+		public bool AppearsInStoryOnly { get; set; } = false;
+		public bool AppearsInSurvivalOnly { get; set; } = false;
+		[JsonConverter(typeof(JsonStringEnumConverter))]
+		public SkillType AppliedSkill { get; set; } = SkillType.None;
+		[JsonConverter(typeof(JsonStringEnumConverter))]
+		public SkillType ImprovedSkill { get; set; } = SkillType.None;
+		[JsonConverter(typeof(JsonStringEnumConverter))]
+		public CraftingLocation RequiredCraftingLocation { get; set; } = CraftingLocation.Anywhere;
+		public bool RequiresLitFire { get; set; } = false;
+		public bool CanIncreaseRepairSkill { get; set; } = false;
+
+		#region Json
+		public static ModUserBlueprintData ParseFromJson(string jsonText)
+		{
+			return JsonSerializer.Deserialize<ModUserBlueprintData>(jsonText);
```

Comment:
`?? throw new ArgumentException("Could not parse blueprint data from the text.", nameof(jsonText))`